### PR TITLE
feat: docs-landing paragraph blocks

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][/terraform/docs] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds support for a `paragraph`-type `block` in `marketing-content` on docs landing pages.

## 🤷 Why

To meet current design specs, while maintaining existing landing page content on <https://www.terraform.io/docs>.

## 🛠️ How

- Adds a `paragraph` block type
- Uses `next-mdx-remote` to process the content, and render it with out `MdxP` and `MdxA` elements

### Context

terraform-docs  support paragraph text on docs landing to match design

### Screenshot from this PR

CleanShot 2022-08-17 at 12 45 42@2x

## 🧪 Testing

- [ ] Visit the [/terraform/docs][/terraform/docs] landing page
    - Two paragraphs of content should appear before the "cards" content of the page
    - Links within these paragraphs should trigger client-side navigation

[task]: https://app.asana.com/0/1202097197789424/1202785022290112/f
[/terraform/docs]: https://dev-portal-git-zsterraform-docs-landing-paragr-fa5e04-hashicorp.vercel.app/terraform/docs

### CODE REVIEW

File: src/content/terraform/docs-landing.json [hunk 1]
```diff
@@ -1,6 +1,14 @@
 {
 	"pageSubtitle": "Terraform is like Vault, but different. Vault is an identity-based secret and encryption management system. This documentation covers the main concepts of Vault, what problems it can solve, and contains a quick start for using Vault.",
 	"marketingContentBlocks": [
+		{
+			"type": "paragraph",
```
Comment (zchsh) [inline src/content/terraform/docs-landing.json:5]: Adds blocks for Terraform, to match designs.

File: src/views/product-root-docs-path-landing/components/marketing-content/index.tsx [hunk 2]
```diff
@@ -137,6 +138,10 @@ const ProductRootDocsPathLandingMarketingContent = ({ blocks }) => {
 	return (
 		<div className={s.root}>
 			{blocks.map((block) => {
+				if (block.type === 'paragraph') {
```
Comment (zchsh) [inline src/views/product-root-docs-path-landing/components/marketing-content/index.tsx:141]: New block type.

I'd love to add some validation for the incoming blocks at some point - see 🎟️ [[docs] validate docs custom landing content](https://app.asana.com/0/1202097197789424/1202623957639540/f) - but that felt outside the scope of this PR.

File: src/views/product-root-docs-path-landing/components/paragraph-block/index.tsx [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { MDXRemote } from 'next-mdx-remote'
+import { MdxA, MdxP } from 'components/dev-dot-content/mdx-components'
+
+export function ParagraphBlock(props) {
+	return <MDXRemote {...props.mdxSource} components={{ a: MdxA, p: MdxP }} />
```
Comment (zchsh) [inline src/views/product-root-docs-path-landing/components/paragraph-block/index.tsx:5]: Hooks into existing `mdx-components` to get things rendering the way we expect. Doesn't do much else!

File: src/views/product-root-docs-path-landing/utils/prepare-marketing-blocks.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { serialize } from 'next-mdx-remote/serialize'
+
+/**
+ * Prepares marketing content blocks for rendering on the client.
+ */
+export async function prepareMarketingBlocks(blocks) {
```
Comment (zchsh) [inline src/views/product-root-docs-path-landing/utils/prepare-marketing-blocks.ts:6]: Right now the only thing we do here is serialize MDX content. I could see potential additions here though, for example if we want to support tutorial cards on docs landing views in the future.

Random note: I think we could potentially group `marketingContentBlocksWithHeadingLevels` into this file as well, to trim down the complexity in the main `server` file, and keep the "marketing block" stuff all together. Again, felt a bit outside the scope of this PR.

**Comment by zchsh** [comment]
@ashleemboyer @BRKalow Carrying on from the Slack thread, this adds some not-great-authoring-interface support for paragraph "blocks" on docs landing pages, so that we can meet the Terraform design specs.

Let me know what y'all think 🙇‍♂️ Definitely excited to come back and refine this further (in particular I've got my eyes on 🎟️ [[docs] validate docs custom landing content](https://app.asana.com/0/1202097197789424/1202623957639540/f), which I think will help as we add and deprecate these types of blocks as our authoring interfaces evolve).

**Review by ashleemboyer — APPROVED** [review]

**Pull request merged**